repos/spack_repo/builtin/packages/py_dask_mpi/package.py

```diff
@@ -32,3 +39,16 @@ class PyDaskMpi(PythonPackage):
     # jupyter-server-proxy is not a needed dependency; https://github.com/dask/dask-mpi/pull/102
     # this significantly reduces the dependency tree of py-dask-mpi
     patch("remove-dependency-jupyter-proxy.patch", when="@:2022.4.0")
+
+    # fix for python 3.12+, merged into mainline
+    # https://github.com/dask/dask-mpi/pull/123
+    conflicts("^python@3.12:", when="@:2022")
```

Comment:
```suggestion
    depends_on("python@:3.11", when="@:2022")
```
This will make it easier for `spack audit packages` to notice when Python 3.11 is removed and the corresponding versions of py-dask-mpi need to be removed.